src/ethereum.js

```diff
@@ -0,0 +1,41 @@
+require('babel-polyfill')
+require('dotenv').config({path: './.env'})
+const { eth, contracts } = require('decentraland-commons')
+const web3Eth = eth
+const { LANDRegistry } = contracts
+
+async function connectBlockchain () {
+  try {
+    await web3Eth.disconnect() // clean if it is a retry
+    let connected = await web3Eth.connect({
+      contracts: [LANDRegistry]
+    })
+    if (!connected) {
+      throw new Error('Could not connect to the blockchain')
+    }
+  } catch (e) {
+    if (e.message.indexOf('Could not connect to the blockchain') !== -1) {
+      console.log('Trying to connect to the blockchain...')
+      setTimeout(connectBlockchain, 3000)
+    }
+  }
+}
+
+async function getIPNS (x, y) {
+  let ipns
+  try {
+    const land = web3Eth.getContract('LANDRegistry')
+    const metadata = await land.landData(x, y)
+    ipns = await LANDRegistry.decodeLandData(metadata).ipns
+    return ipns ? ipns.split(':')[1] : 0
+  } catch (e) {
+    // return undefined ipns
+    console.log(e.message)
+  }
+  return ipns
```

Comment:
?? this returning `undefined` is un clear, that `let ipns` looks weird outside the try/catch, can lead to code smell, is that necessary? if you really need it, you can a finally statement and return `undefined` or `null` prefer.

Reply:
We could return `null` at `catch` clause.